### PR TITLE
#5 Add Alert For Duplicates

### PR DESCRIPTION
**Pull request by Kanishka3**

### CODE REVIEW

File: VocabularyTrainer/AddWord/AddNewWordViewController.swift [hunk 1]
```diff
@@ -83,6 +83,16 @@ final class AddNewWordViewController: UIViewController {
     private var translationHasText = false
     private var completed: (() -> Void)?
     
+    var hasDuplicates : Bool {
```
Comment (misteu) [inline VocabularyTrainer/AddWord/AddNewWordViewController.swift:86]:
```suggestion
    var hasDuplicates: Bool {
```

File: VocabularyTrainer/AddWord/AddNewWordViewController.swift [hunk 2]
```diff
@@ -83,6 +83,16 @@ final class AddNewWordViewController: UIViewController {
...
+        let word = newWordTextField.text ?? ""
+        guard let selectedLanguage = selectedLanguage,
+              let dictionary = UserDefaults.standard.dictionary(forKey: selectedLanguage) as? [String: String] else {
+                  return false
+              }
+        let matches = dictionary.map{String($0.key) }.filter({ $0.lowercased().elementsEqual(word.lowercased()) })
```
Comment (misteu) [inline VocabularyTrainer/AddWord/AddNewWordViewController.swift:92]: Please split add some linebreaks
```suggestion
        let matches = dictionary
          .map { String($0.key) }
          .filter { $0.lowercased().elementsEqual(word.lowercased()) }
```

File: VocabularyTrainer/AddWord/AddNewWordViewController.swift [hunk 3]
```diff
@@ -183,7 +193,7 @@ final class AddNewWordViewController: UIViewController {
         dismiss(animated: true)
     }
     
-    @objc private func addNewWordTapped() {
+    func addWordToDictionary(){
```
Comment (misteu) [inline VocabularyTrainer/AddWord/AddNewWordViewController.swift:196]:
```suggestion
    func addWordToDictionary() {
```

File: VocabularyTrainer/AddWord/AddNewWordViewController.swift [hunk 4]
```diff
@@ -210,12 +220,24 @@ final class AddNewWordViewController: UIViewController {
...
         self.delegate?.wordAdded()
         showToast(message: NSLocalizedString("New word added", comment: "New word added"), yCoord: 340.0)
     }
     
+    @objc private func addNewWordTapped() {
+        hasDuplicates ? showAlert() : addWordToDictionary()
+    }
+    
+    func showAlert() {
+        let duplicateWordAlert = UIAlertController(title: "Duplicate Word Detected", message: "This word already exists in the word list", preferredStyle: .alert)
```
Comment (misteu) [inline VocabularyTrainer/AddWord/AddNewWordViewController.swift:233]: Please use `NSLocalizedString` for all the strings.

I think it makes sense to include the word itself in the alert's message as well. Also we should mention that an existing word will be overwritten when using tap anyway.

Can you change it to something like:
"The word XYZ already exists. Adding the new pair will overwrite the old pair of words."

**Comment by Kanishka3** [comment]
@misteu Done with the changes. Feel free to merge!

**Comment by misteu** [comment]
> @misteu Done with the changes. Feel free to merge! 

Thank you! I will have a look tomorrow :)

**Comment by misteu** [comment]
Alright, thanks @Kanishka3  :) !

**Pull request merged**